src/Auth/AuthSelectionMiddleware.php

```diff
@@ -86,21 +96,61 @@ public function __invoke(CommandInterface $command)
             }
 
             try {
-                $selectedAuthScheme = $resolver->selectAuthScheme(
+                $authSchemeList = $this->buildAuthSchemeList(
                     $resolvableAuth,
+                    $command['@context']['auth_scheme_preference']
+                        ?? null,
+                );
+                $selectedAuthScheme = $resolver->selectAuthScheme(
+                    $authSchemeList,
                     ['unsigned_payload' => $unsignedPayload]
                 );
-            } catch (UnresolvedAuthSchemeException $e) {
+
+                if (!empty($selectedAuthScheme)) {
+                    $command['@context']['signature_version'] = $selectedAuthScheme;
+                }
+            } catch (UnresolvedAuthSchemeException $ignored) {
                 // There was an error resolving auth
                 // The signature version will fall back to the modeled `signatureVersion`
                 // or auth schemes resolved during endpoint resolution
             }
-
-            if (!empty($selectedAuthScheme)) {
-                $command['@context']['signature_version'] = $selectedAuthScheme;
-            }
         }
 
         return $nextHandler($command);
     }
+
+    /**
+     * Prioritizes auth schemes according to user preference order.
+     * User-preferred schemes that are available will be placed first,
+     * followed by remaining available schemes.
+     *
+     * @param array $resolvableAuthSchemeList Available auth schemes
+     * @param array|null $commandAuthSchemePreference Command-level preferences (overrides config)
+     *
+     * @return array Reordered auth schemes with user preferences first
+     */
+    private function buildAuthSchemeList(
```

Comment:
Something like this would be simpler:

```php
private function buildAuthSchemeList(
    array $resolvableAuthSchemeList,
    ?array $commandConfiguredAuthSchemes,
): array {
    $configured = $commandAuthSchemePreference ?? $this->configuredAuthSchemes;

    if (empty($configured)) {
        return $resolvableAuthSchemeList;
    }

    $prioritized = array_intersect($configured, $resolvableAuthSchemeList);
    
    $remaining = array_diff($resolvableAuthSchemeList, $prioritized);

    return array_merge($prioritized, $remaining);
}